yellowbrick/style/colors.py

```diff
@@ -87,15 +87,45 @@ def resolve_colors(n_colors=None, colormap=None, colors=None):
 
     # Work with the colormap if specified and colors is not
     if colormap is not None and colors is None:
+        # Must import here to avoid recursive import
+        from .palettes import PALETTES, ColorPalette
         if isinstance(colormap, str):
             try:
-                colormap = cm.get_cmap(colormap)
+
+                # try to get colormap from PALETTES first
+                _colormap = PALETTES.get(colormap, None)
+
+                if _colormap is None:
+
+                    colormap = cm.get_cmap(colormap)
+                    n_colors = n_colors or len(get_color_cycle())
+                    _colors = list(map(colormap, np.linspace(0, 1, num=n_colors)))
+
+                else:
+
+                    _colors = ColorPalette(_colormap).as_rgb()
+                    n_colors = n_colors or len(_colors)
+
             except ValueError as e:
+
                 raise YellowbrickValueError(e)
 
+        # if yellowbrick color palette is provided as colormap
+        elif isinstance(colormap, ColorPalette):
 
-        n_colors = n_colors or len(get_color_cycle())
-        _colors = list(map(colormap, np.linspace(0, 1, num=n_colors)))
+            _colors = colormap.as_rgb()
+            n_colors = n_colors or len(_colors)
+
+        # if matplotlib color palette is provided as colormap
+        elif isinstance(colormap, mpl.colors.Colormap):
+            n_colors = n_colors or len(get_color_cycle())
+            _colors = list(map(colormap, np.linspace(0, 1, num=n_colors)))
+        else:
+            raise TypeError(
+                "Colormap type %s is not recognized. Possible types are: %s"
+                % (type(colormap), ''.join(['yellowbrick.style.ColorPalette',
+                                            'matplotlib.cm',
+                                            'str'])))
```

Comment:
Thank you for adding this helpful error message! Two small requests &mdash; would you please change this to be a more Yellowbrick-specific `YellowbrickValueError` instead of a `TypeError`, and would you mind changing to the [new string formatting style](https://realpython.com/python-string-formatting/#2-new-style-string-formatting-strformat) (YB examples [here](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/features/manifold.py#L242-L244) and [here](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/features/jointplot.py#L149-L152))?

Reply:
sorry for having so many careless mistakes with the code. I hope it is satisfactory now.

Reply:
No trouble at all! Just part of the process of getting up to speed as a new contributor :)